src/IceRpc/Transports/Slic/Internal/SlicConnection.cs

```diff
@@ -598,6 +571,47 @@ async Task PerformDisposeAsync()
             _nextBidirectionalId = 0;
             _nextUnidirectionalId = 2;
         }
+
+        void SendPing(long payload)
+        {
+            try
+            {
+                WriteConnectionFrame(FrameType.Ping, new PingBody(payload).Encode);
+            }
+            catch (IceRpcException)
+            {
+                // Expected if the connection is closed.
+            }
+            catch (Exception exception)
+            {
+                Debug.Fail($"The sending of a Ping frame failed with an unexpected exception: {exception}");
+                throw;
+            }
+        }
+
+        void SendReadPing()
+        {
+            // ReadKeepAlive is no-op unless _pendingPongCount == 0 before the Increment below: we send a Ping only if
+            // there is no outstanding Pong.
+            if (Interlocked.Increment(ref _pendingPongCount) == 1)
+            {
+                SendPing(1L);
+            }
+            else
+            {
+                Interlocked.Decrement(ref _pendingPongCount);
```

Comment:
Can you explain this? It is not obvious to me why the call to SendReadPing decrements the pong count here.

Reply:
```csharp
           if (Interlocked.Increment(ref _pendingPongCount) == 1)
            {
                SendPing(1L);
            }
            else
            {
                Interlocked.Decrement(ref _pendingPongCount);
            }
```

The code is quite simple. 

1 - we always increment _pendingPongCount
2 - if the incremented value is 1 -- meaning there was no pending Pong -- we send a Ping. The _pendingPongCount will be decremented when we receive the Pong
3 - if the incremented value has a different value, we decremented it to make the whole local function overall no-op

Reply:
We can do the same with `CompareExchange` without increment/decrement
```
if (Interlocked.CompareExchange(ref _pendingPongCount, 1, 0) == 0)
{
    SendPing(1L);
}
```

Reply:
Updated as suggested